admin/testbrew.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Script executed by Buildbot on the Mac mini build slave, this instantiates the
+OS X Virtual Machine to a snapshot with a clean homebrew install, then executes
+via SSH the brew update, install and test commands to verify a successful
+installation of a homebrew recipe.
+
+The host machine must have:
+        * A VMWare OS X VM available at a particular location
+
+The guest virtual machine must have:
+        * A snapshot with a particular name, homebrew-clean
+        * Homebrew installed and available
+"""
+
+import os
+import sys
+import urllib2
+
+from twisted.python.usage import Options, UsageError
+
+from subprocess import check_output, CalledProcessError
+
+from flocker.provision._install import run_with_fabric, task_test_homebrew
+
+
+YOSEMITE_VMX_PATH = (
+    '"'
+    "{home}/Documents/Virtual Machines.localized/"
+    "OS X 10.10.vmwarevm/OS X 10.10.vmx"
+    '"'
+).format(home=os.getenv("HOME"))
+
+YOSEMITE_SNAPSHOT = "homebrew-clean"
+
+VM_USERNAME = "ClusterHQVM"
+
+VM_HOST = "10.0.126.88"
+
+
+class TestBrewOptions(Options):
+    description = ("Start a VMWare OS X VM and test a brew recipe "
+                   "installation.")
+
+    optParameters = [
+        ['vmhost', 'h', VM_HOST,
+         'IP address or hostname of the Virtual Machine'],
+        ['vmuser', 'u', VM_USERNAME,
+         'Username for the Virtual Machine'],
+        ['vmpath', 'p', YOSEMITE_VMX_PATH,
+         'Full path to the Virtual Machine image'],
+        ['vmsnapshot', 's', YOSEMITE_SNAPSHOT,
+         'Snapshot identifier for the Virtual Machine'],
+    ]
+
+    synopsis = ('Usage: test-brew-recipe <recipe URL> [--vmhost <host>] '
+                '[--vmuser <username>] '
+                '[--vmpath <path>] '
+                '[--vmsnapshot <name>]')
```

Comment:
The typical idiom is `[options]`.  `Options` will automatically generate help text for the various options that actually exist.  This way there's only one version of truth to maintain.